### DISCUSSION TITLE
"cross-browser compatibility" styles not working as expected on safari browser

### DISCUSSION THREAD

**Discussion by akram-hossan** [tailwindlabs/tailwindcss#12620] [Help] [answered]
### What version of Tailwind CSS are you using?

v3.1.8

### What build tool (or framework if it abstracts the build tool) are you using?

next v12.2.5

### What browser are you using?

Chrome, Firefox, Safari, or N/A

### What operating system are you using?

Windows / MacOS

### Reproduction URL

https://play.tailwindcss.com/HLvEQaBzPF  

### Problem description:  In my current Project I am facing an issue. In web or Android, my style works perfectly. But when I open my application on Safari browser from Mac or iPhone it doesn't produce the expected output. 

Here are some screenshots for better understanding: 

### Expected output: 
![image](https://github.com/tailwindlabs/tailwindcss/assets/127820204/2935b2a1-2a73-4ea2-aac6-567a7f60c5bc)

But on the Safari browser both from Mac or iPhone we are getting this output: 
![image](https://github.com/tailwindlabs/tailwindcss/assets/127820204/ffed23a6-722c-4087-9c67-2990b8612db2)

```javascript
 <div className=" flex flex-col lg:flex-row w-full h-full">
       <div className="lg:w-80 w-full lg:flex lg:flex-col grid grid-cols-3  "> <div>
        <div className="w-full h-auto px-7 py-5 pt-10 relative"><div>
<div>
```
This code block is not working properly on the Safari browser on the mobile screen.

Source code: https://gist.github.com/akram-hossan/d3d041d375d06b5165a5eace002c6b29

**Comment by fordaaron** [comment, marked as answer]
Removing `h-full` from the following element will resolve the issue. This is due to a difference in how Safari handles height percentages.
```diff
<div
  key={item.id}
- className={`lg:px-3 xl:px-6 md:px-[2.29rem] px-3 lg:py-14 py-8 h-full border border-neutral- focus/[.3] border-solid ${
+ className={`lg:px-3 xl:px-6 md:px-[2.29rem] px-3 lg:py-14 py-8 border border-neutral- focus/[.3] border-solid ${
    item.isActive ? 'bg-[#68749B]' : 'bg-transparent'
  }`}
  onClick={() => handleTransactionClick(item.id)}
  style={{ cursor: !item.isActive ? 'pointer' : 'default' }}>
```

https://play.tailwindcss.com/6vL4L7UgCf

**Reply by akram-hossan** [reply]
Thank You so much

**Reply by Chickenroast** [reply]
Thank you guys for existing <3
